Elk.py

```diff
@@ -0,0 +1,348 @@
+def send_json_to_ELK(file_name, index_name, instance_id, time, account_id, session_id, type_of, es):
+    """
+    file got to be in ndjson format
+    """
+    try:
+        with open(file_name) as fp:
+            for line in fp:
+                line = line.replace("\n", "")
+                # line = line.replace(" ", "")
+                line = line.strip()
+                # jdoc = {"hostname": hostname, "ipaddr": ipaddr, "type": type_of, "data": json.loads(line)}
+                if type_of != "lynis":
+                    jdoc = {"instance_id": instance_id, "time": time, "account_id": account_id,
+                            "session_id": session_id,
+                            "type_of_scan": type_of, "data": json.loads(line)}
+                else:
+                    # lynis:
+                    # need to break the title thing:
+                    json_line = json.loads(line)
+                    title = json_line["title"]
+                    json_line.pop("title")
+                    jdoc = {"instance_id": instance_id, "time": time, "account_id": account_id,
+                            "session_id": session_id,
+                            "type_of_scan": type_of, "title": title, title: json_line}  # about "title":title
+                es.index(index=index_name, doc_type='_doc', body=jdoc)
+
+        print("finished upload: " + index_name)
+    except Exception as e:
+        print(e)
+
+
+def main():
+    tprint("ELK    EC2    SCAN")
+    link = input("insert your Elk URL (e.g: localhost:9200) : ")
+    username = input("insert your Elk username for auth(if there is no auth, click ENTER): ")
+    # (?) need to get the directories too (vuls and lynis), sudo password
+
+    if username != "":
+        password = input("insert your Elk password for auth: ")
+        elastic = Elasticsearch([link], http_auth=(username, password))
+    else:
+        elastic = Elasticsearch([link])
+    print("")
+    print("running ...")
+
+    # get IP:
+    hostname = socket.gethostname()
+    local_ip = socket.gethostbyname(hostname)
+
+    sudo_pass = ''  # sudo password of the machine
+    sudo_password = "echo " + sudo_pass + " | sudo -S "
+
+    begin_time = datetime.datetime.now()
+
+    vuls_directory = "home/ubuntu/idannos"
+    lynis_directory = "home/ubuntu/lynis"  # don't really use it the function, explanation there.
+
+    chkrotkit(sudo_password)
+    print("finished rootkit")
+
+    vuls(vuls_directory, sudo_password)
+    print("finished vuls")
+
+    lynis(lynis_directory, sudo_password)
+    print("finished lynis")
+
+    # get the public IP:
+    public_ip = str(get('https://api.ipify.org').text)
+
+    # get the instance_id:
+    instance_id = ec2_metadata.instance_id
+    # uuid:
+    uuid_string = str(uuid.uuid4())
+
+    # timeStamp:
+    date_and_hour = datetime.datetime.now()
+    temp = str(date_and_hour).split(" ")
+    date = temp[0]  # getting the date only without hours
+
+
+    # need to fill this before running:
```

Comment:
boto3 has a built-in default to read these arguments from environment variables if set or read them from user input